### PR TITLE
Check DeprecationWarning in Variable

### PR DESCRIPTION
**Pull request by unnonouno**
I inserted `assertiWarnig` to suppress warning in unit test.

### CODE REVIEW

**Comment by unnonouno** [comment]
assertWarns is only available in py3.

**Comment by niboshi** [comment]
Instead of ignoring, how about checking the specified warning is actually emitted?

As you specified `record=True`, `warnings.catch_warnings` returns a list of `WarningMessage`s actually emitted. You can check their `category` attributes.

**Comment by unnonouno** [comment]
Actually it does not correctly work in py2.

**Comment by niboshi** [comment]
What do you mean? In my test it worked in both python 2 and 3.

```
import warnings

with warnings.catch_warnings(record=True) as w:
    warnings.simplefilter('always')
    warnings.warn('hello', DeprecationWarning)
    print([_.category for _ in w])
```
Python 3
```
[<class 'DeprecationWarning'>]
```
Python 2
```
[<type 'exceptions.DeprecationWarning'>]
```

**Comment by unnonouno** [comment]
Maybe this is a bug of python2. When a concrete warning class is set as `module`, that means the warning is shown  once in a module, `warnings` module stores this exception class to ignore next time. After that this warning class is not shown even when `simplefilter('always')` is applied. Maybe it is because the list in `warnings` stores this warning class and it is ignored.
Actually when I used `catch_warnings`, test is not passed.

**Comment by niboshi** [comment]
Sorry, what does this mean? Can you provide a code fragment?

>When a concrete warning class is set as `module`,

**Comment by niboshi** [comment]
I confirmed this code does not catch warnings in Python 2.

```
import warnings

def f():
    warnings.warn('hello', DeprecationWarning)


def fff():
    with warnings.catch_warnings(record=True) as w:
        warnings.simplefilter('always')
        f()
    print([_.category for _ in w])


f()
fff()
```
Python 2
```
[]
```

**Comment by niboshi** [comment]
From the result above, I guess this problem would not happen if all `DeprecationWarning`s were correctly wrapped with `assert_warns` (or avoid using methods that emit the warning).

How do you think?

**Comment by unnonouno** [comment]
I'll try it.

**Comment by unnonouno** [comment]
As you say, it passes the test.

File: chainer/testing/helper.py [hunk 1]
```diff
@@ -29,3 +31,22 @@ def with_requires(*requirements):
...
+@contextlib.contextmanager
+def assert_warns(expected):
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('always')
+        yield
+
+    for m in w:
+        if not isinstance(m.message, expected):
+            continue
+        return
```
Comment (niboshi) [inline chainer/testing/helper.py]: How about simply `if not any(isinstance(m.message, expected) for m in w)`?

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**